File: backend/scripts/createAdmin.js
require('dotenv').config();
const mongoose = require('mongoose');
const User = require('../src/models/User');
const logger = require('../src/utils/logger');

const connectDB = async () => {
    try {
        await mongoose.connect(process.env.MONGODB_URI);
        logger.info('✅ MongoDB Connected');
    } catch (error) {
        logger.error('❌ MongoDB connection failed:', error);
        process.exit(1);
    }
};

const createAdmin = async () => {
    try {
        await connectDB();

        const adminEmail = process.env.DEFAULT_ADMIN_EMAIL || 'admin@yolo.com';
        const adminPassword = process.env.DEFAULT_ADMIN_PASSWORD || 'Admin123!';
        const adminName = process.env.DEFAULT_ADMIN_NAME || 'Super Admin';

        // Check if admin exists
        const existingAdmin = await User.findOne({ emailUser: adminEmail });

        if (existingAdmin) {
            logger.info('⚠️  Admin user already exists');
            logger.info(`Email: ${adminEmail}`);
            process.exit(0);
        }

        // Create admin
        const admin = await User.create({
            namaUser: adminName,
            emailUser: adminEmail,
            passwordUser: adminPassword,
            role: 'admin',
            isActive: true
        });

        logger.info('✅ Admin user created successfully!');
        logger.info('═══════════════════════════════════');
        logger.info(`Email: ${adminEmail}`);
        logger.info(`Password: ${adminPassword}`);
        logger.info('═══════════════════════════════════');
        logger.info('⚠️  Please change the password after first login!');

        process.exit(0);

    } catch (error) {
        logger.error('❌ Error creating admin:', error);
        process.exit(1);
    }
};

createAdmin();

File: backend/scripts/create_admin.py
"""
Script to create admin user
Run: python scripts/create_admin.py
"""

import asyncio
import os
import sys
from datetime import datetime

# Add parent directory to path
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from dotenv import load_dotenv
load_dotenv()

from motor.motor_asyncio import AsyncIOMotorClient
from app.utils.password import hash_password


async def create_admin():
    """Create admin user"""
    
    # Get MongoDB URI
    mongodb_uri = os.getenv("MONGODB_URI")
    if not mongodb_uri:
        print("❌ MONGODB_URI not set in environment variables")
        return
    
    # Connect to MongoDB
    client = AsyncIOMotorClient(mongodb_uri)
    db = client[os.getenv("DB_NAME", "yolo_detection")]
    
    # Admin details
    admin_data = {
        "namaUser": "Administrator",
        "emailUser": "admin@admin.com",
        "passwordUser": hash_password("admin123"),
        "role": "admin",
        "isActive": True,
        "createdAt": datetime.utcnow(),
        "updatedAt": datetime.utcnow()
    }
    
    # Check if admin exists
    existing = await db.users.find_one({"emailUser": admin_data["emailUser"]})
    if existing:
        print(f"⚠️ Admin dengan email {admin_data['emailUser']} sudah ada")
        client.close()
        return
    
    # Create admin
    result = await db.users.insert_one(admin_data)
    
    print("✅ Admin berhasil dibuat!")
    print(f"   ID: {result.inserted_id}")
    print(f"   Email: {admin_data['emailUser']}")
    print(f"   Password: admin123")
    print("\n⚠️ Segera ganti password setelah login!")
    
    client.close()


if __name__ == "__main__":
    asyncio.run(create_admin())
